src/saving/SaveHelper.cs

```diff
@@ -22,9 +22,52 @@ public enum SaveOrder
         FileSystem,
     }
 
+    public static void RemoveExcessAutoSaves()
+    {
+        int maxAutoSaves = Settings.Instance.MaxAutoSaves;
+
+        var autoSaveNames = new List<string>();
+        var allSaveNames = CreateListOfSaves();
+
+        foreach (var save in allSaveNames.AsEnumerable().Reverse())
+        {
+            if (save.StartsWith("auto_save", StringComparison.CurrentCulture))
+            {
+                autoSaveNames.Add(save);
+            }
+
+            if (autoSaveNames.Count >= maxAutoSaves && autoSaveNames.Count > 0)
```

Comment:
I think this if is wrong. If I set autosaves to keep to 1 then all autosaves are deleted.